### PR TITLE
fix: add validation to patching project key (ET-305)

### PR DESCRIPTION
**Pull request by corban-beaird**
## Ticket
ET-305

## Description

Adds validation on project keys when making PatchProject requests. 

## Test Plan

- Automated test pass (`TestPatchProjectWithInvalidProjectKey`)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/project/postgres_project.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ const (
 	// MaxRetries is the maximum number of retries for transaction conflicts.
 	MaxRetries = 5
 	// ProjectKeyRegex is the regex pattern for a project key.
-	ProjectKeyRegex = "^[A-Z0-9]{1,5}$"
+	ProjectKeyRegex = "^[a-zA-Z0-9]{1,5}$"
```
Comment (AmanuelAaron) [inline master/internal/project/postgres_project.go:32]: Do we want to allow lower case keys? Does this now mean that keys are case sensitive? My original thought was that we just capitalize the characters

Comment (corban-beaird) [reply master/internal/project/postgres_project.go:32]: Keys are case insensitive. I changed the validation to allow for folks to submit keys in whatever case they'd like, it gets stored & displayed in uppercase.

But I see you're point if someone is using mixed cases to submit a key that could be confusing.

Comment (corban-beaird) [reply master/internal/project/postgres_project.go:32]: This is a good point in terms of understanding where the responsibility for preparing this information should lie. Is it the responsibility of the request to make sure that the keys are in uppercase (i.e. before sending the request, capitalizing the values or should be be done in `postgres_project.go` prior to querying the db).

I had been of the mindset to allow the requested key to come in as any case; however, that could be misunderstood as allowing case sensitive keys. 

Based on that, I think it makes more sense to require that the request to be responsible for ensuring that the incoming key is in the correct format & returning an error if it's not so it's abundantly clear that keys are required to be capitalized.

Comment (AmanuelAaron) [reply master/internal/project/postgres_project.go:32]: Ahh I see so this is just for validation. I think that is fine then

**Review by AmanuelAaron — APPROVED** [review]
LGTM

File: master/internal/project/postgres_project.go (hunk 1 above)
Comment (AmanuelAaron) [reply master/internal/project/postgres_project.go:32]: Maybe a warning message perhaps. But it may just be easier to force capital letters

Comment (corban-beaird) [reply master/internal/project/postgres_project.go:32]: I think even just having some sort of note that users can see in the project creation/edit modal next to the text field for the project key letting them know that it's case insensitive will be the best of both worlds

**Pull request merged**